3.resistance-signature/scripts/nbconverted/0.training-test-split.py

```diff
@@ -267,56 +267,57 @@
 # In[12]:
 
 
+### Remove Batch 3 ###
 # Load inference data (a different hold out)
-inference_batch = "2019_06_25_Batch3"
-inference_file = pathlib.Path(f"../3.bulk-signatures/data/{inference_batch}_combined_normalized.csv.gz")
-inference_df = pd.read_csv(inference_file)
+#inference_batch = "2019_06_25_Batch3"
+#inference_file = pathlib.Path(f"../3.bulk-signatures/data/{inference_batch}_combined_normalized.csv.gz")
+#inference_df = pd.read_csv(inference_file)
 
-inference_df = inference_df.assign(
-    Metadata_dataset="untreated_mystery_clones",
-    Metadata_batch=inference_batch,
-    Metadata_clone_type="resistant",
-    Metadata_clone_type_indicator=1,
-    Metadata_model_split="inference"
-)
+#inference_df = inference_df.assign(
+    #Metadata_dataset="untreated_mystery_clones",
+    #Metadata_batch=inference_batch,
+    #Metadata_clone_type="resistant",
+    #Metadata_clone_type_indicator=1,
+    #Metadata_model_split="inference"
+#)
 
-inference_df.loc[inference_df.Metadata_clone_number.str.contains("WT"), "Metadata_clone_type"] = "sensitive"
-inference_df.loc[inference_df.Metadata_clone_number.str.contains("WT"), "Metadata_clone_type_indicator"] = 0
-inference_df = inference_df.assign(
-    Metadata_unique_sample_name=[f"profile_{x}_inference" for x in range(0, inference_df.shape[0])]
-)
+#inference_df.loc[inference_df.Metadata_clone_number.str.contains("WT"), "Metadata_clone_type"] = "sensitive"
+#inference_df.loc[inference_df.Metadata_clone_number.str.contains("WT"), "Metadata_clone_type_indicator"] = 0
+#inference_df = inference_df.assign(
+#    Metadata_unique_sample_name=[f"profile_{x}_inference" for x in range(0, inference_df.shape[0])]
+#)
 
-inference_df.Metadata_clone_number.value_counts()
+#inference_df.Metadata_clone_number.value_counts()
 
 
 # In[13]:
 
 
 # Combine profiles into a single dataset and output
-bortezomib_df = pd.concat(
-    [
-        full_df.query("Metadata_dataset == 'bortezomib'"),
-        inference_df
-    ],
-    axis="rows",
-    sort=False
-).reset_index(drop=True)
+#bortezomib_df = pd.concat(
+ #   [
+  #      full_df.query("Metadata_dataset == 'bortezomib'"),
+   #     inference_df
+    #],
+    #axis="rows",
+    #sort=False
+#).reset_index(drop=True)
 
-output_file = pathlib.Path(f"{output_dir}/bortezomib_signature_analytical_set.tsv.gz")
-bortezomib_df.to_csv(output_file, sep="\t", index=False)
+#output_file = pathlib.Path(f"{output_dir}/bortezomib_signature_analytical_set.tsv.gz")
+#bortezomib_df.to_csv(output_file, sep="\t", index=False)
```

Comment:
We actually still want this to output a file called `bortezomib_signature_analytical_set.tsv.gz` I believe.

So, the way to change this (make sure you change it in the `.ipynb` file and not in the nbconverted `.py` file) is the following:

```python
# Drop the inference data
bortezomib = full_df.query("Metadata_dataset == 'bortezomib'").reset_index(drop=True)

# Output file
output_file = pathlib.Path(f"{output_dir}/bortezomib_signature_analytical_set.tsv.gz")
bortezomib_df.to_csv(output_file, sep="\t", index=False)
```

Alternatively, since this is technically an enhancement and not a _required_ change. You can do the following (and this is always available to you as an option to address a pull request comment)

Create an issue here: https://github.com/broadinstitute/profiling-resistance-mechanisms/issues that links to this pull request and specifically references the file that you need to change and how you will change it.

For instance: "In the notebook `3.resistance-signature/0.training-test-split.ipynb` (add link) we dropped the `inference set` (batch 3) because of overly confluent plates and suboptimal plate design. In the notebook, I still need to output the bortezomib signature analytical set, but I can also include the new batches of data, which will serve as a better experimentally designed inference set as included in pull request (use a `#` to reference all relevant issues/PRs)"

Reply:
this comment ☝️ is still valid (not outdated)